caliper@2.8.0 +rocm fails on OLCF Crusher w/ PrgEnv-gnu: libcaliper-serial.so.2.8.0: undefined reference to hipKernelNameRef

### Steps to reproduce the issue

I'm having trouble building `caliper@2.8.0 +rocm` with any of the following PrgEnv on OLCF Crusher:
* `PrgEnv-gnu`

In all cases, I'm trying to use `rocm/5.2.0` defined via externals in `packages:` section.

<details><summary>Spack environment</summary><pre>
spack:
  view: false
  concretizer:
    reuse: false
    unify: false
  compilers:
  - compiler:
      spec: gcc@11.2.0
      paths:
        cc: gcc
        cxx: g++
        f77: gfortran
        fc: gfortran
      flags: {}
      operating_system: sles15
      target: any
      modules:
      - PrgEnv-gnu
      - gcc/11.2.0
      environment: {}
      extra_rpaths: []
  packages:
    all:
      compiler: [gcc@11.2.0]
      providers:
        blas: [openblas]
        lapack: [openblas]
        mpi: [cray-mpich]
        scalapack: [netlib-scalapack]
      target: [zen3]
      variants: +mpi
    binutils:
      variants: +ld +gold +headers +libiberty ~nls
    elfutils:
      variants: +bzip2 ~nls +xz
    hdf5:
      variants: +fortran +hl +shared
    libfabric:
      buildable: false
      externals:
      - spec: libfabric@1.15
        modules:
        - libfabric/1.15.0.0

    # ROCm 5.2.0
    libunwind:
      variants: +pic +xz
    mpich:
      variants: ~wrapperrpath
    ncurses:
      variants: +termlib
    openblas:
      variants: threads=openmp
    python:
      version: [3.8.13]
    trilinos:
      variants: +amesos +amesos2 +anasazi +aztec +belos +boost +epetra +epetraext
        +ifpack +ifpack2 +intrepid +intrepid2 +isorropia +kokkos +ml +minitensor +muelu
        +nox +piro +phalanx +rol +rythmos +sacado +stk +shards +shylu +stokhos +stratimikos
        +teko +tempus +tpetra +trilinoscouplings +zoltan +zoltan2 +superlu-dist gotype=long_long
    xz:
      variants: +pic
    mesa:
      version: [21.3.8]
    petsc:
      variants: +batch
    parallel-netcdf:
      variants: +cxx +fortran +fpic
    netlib-scalapack:
      variants: +fpic
    netcdf-c:
      variants: ~hdf4 +mpi +parallel-netcdf +shared
    fftw:
      variants: precision=float,double,long_double
    darshan-runtime:
      variants: scheduler=slurm
    cray-mpich:
      buildable: false
      externals:
      - spec: cray-mpich@8.1.16 %gcc@11.2.0
        prefix: /opt/cray/pe/mpich/8.1.16/ofi/gnu/9.1
        modules:
        - cray-mpich/8.1.16
        - libfabric/1.15.0.0
    comgr:
      buildable: false
      externals:
      - spec: comgr@5.2.0
        prefix: /opt/rocm-5.2.0/
        modules: [rocm/5.2.0]
    hip-rocclr:
      buildable: false
      externals:
      - spec: hip-rocclr@5.2.0
        prefix: /opt/rocm-5.2.0/hip
        modules: [rocm/5.2.0]
    hipblas:
      buildable: false
      externals:
      - spec: hipblas@5.2.0
        prefix: /opt/rocm-5.2.0/
        modules: [rocm/5.2.0]
    hipcub:
      buildable: false
      externals:
      - spec: hipcub@5.2.0
        prefix: /opt/rocm-5.2.0/
        modules: [rocm/5.2.0]
    hipfft:
      buildable: false
      externals:
      - spec: hipfft@5.2.0
        prefix: /opt/rocm-5.2.0/
        modules: [rocm/5.2.0]
    hipsparse:
      buildable: false
      externals:
      - spec: hipsparse@5.2.0
        prefix: /opt/rocm-5.2.0/
        modules: [rocm/5.2.0]
    miopen-hip:
      buildable: false
      externals:
      - spec: hip-rocclr@5.2.0
        prefix: /opt/rocm-5.2.0/
        modules: [rocm/5.2.0]
    miopengemm:
      buildable: false
      externals:
      - spec: miopengemm@5.2.0
        prefix: /opt/rocm-5.2.0/
        modules: [rocm/5.2.0]
    rccl:
      buildable: false
      externals:
      - spec: rccl@5.2.0
        prefix: /opt/rocm-5.2.0/
        modules: [rocm/5.2.0]
    rocblas:
      buildable: false
      externals:
      - spec: rocblas@5.2.0
        prefix: /opt/rocm-5.2.0/
        modules: [rocm/5.2.0]
    rocfft:
      buildable: false
      externals:
      - spec: rocfft@5.2.0
        prefix: /opt/rocm-5.2.0/
        modules: [rocm/5.2.0]
    rocm-clang-ocl:
      buildable: false
      externals:
      - spec: rocm-clang-ocl@5.2.0
        prefix: /opt/rocm-5.2.0/
        modules: [rocm/5.2.0]
    rocm-cmake:
      buildable: false
      externals:
      - spec: rocm-cmake@5.2.0
        prefix: /opt/rocm-5.2.0/
        modules: [rocm/5.2.0]
    rocm-dbgapi:
      buildable: false
      externals:
      - spec: rocm-dbgapi@5.2.0
        prefix: /opt/rocm-5.2.0/
        modules: [rocm/5.2.0]
    rocm-debug-agent:
      buildable: false
      externals:
      - spec: rocm-debug-agent@5.2.0
        prefix: /opt/rocm-5.2.0/
        modules: [rocm/5.2.0]
    rocm-device-libs:
      buildable: false
      externals:
      - spec: rocm-device-libs@5.2.0
        prefix: /opt/rocm-5.2.0/
        modules: [rocm/5.2.0]
    rocm-gdb:
      buildable: false
      externals:
      - spec: rocm-gdb@5.2.0
        prefix: /opt/rocm-5.2.0/
        modules: [rocm/5.2.0]
    rocm-opencl:
      buildable: false
      externals:
      - spec: rocm-opencl@5.2.0
        prefix: /opt/rocm-5.2.0/opencl
        modules: [rocm/5.2.0]
    rocm-smi-lib:
      buildable: false
      externals:
      - spec: rocm-smi-lib@5.2.0
        prefix: /opt/rocm-5.2.0/
        modules: [rocm/5.2.0]
    hip:
      buildable: false
      externals:
      - spec: hip@5.2.0
        prefix: /opt/rocm-5.2.0/hip
        modules:
        - rocm/5.2.0
        extra_attributes:
          compilers:
            c: /opt/rocm-5.2.0/llvm/bin/clang++
            c++: /opt/rocm-5.2.0/llvm/bin/clang++
            hip: /opt/rocm-5.2.0/hip/bin/hipcc
    llvm-amdgpu:
      buildable: false
      externals:
      - spec: llvm-amdgpu@5.2.0
        prefix: /opt/rocm-5.2.0/llvm
        modules: [rocm/5.2.0]
        extra_attributes:
          compilers:
            c: /opt/rocm-5.2.0/llvm/bin/clang++
            cxx: /opt/rocm-5.2.0/llvm/bin/clang++
    hsakmt-roct:
      buildable: false
      externals:
      - spec: hsakmt-roct@5.2.0
        prefix: /opt/rocm-5.2.0/
        modules: [rocm/5.2.0]
    hsa-rocr-dev:
      buildable: false
      externals:
      - spec: hsa-rocr-dev@5.2.0
        prefix: /opt/rocm-5.2.0/
        modules: [rocm/5.2.0]
        extra_atributes:
          compilers:
            c: /opt/rocm-5.2.0/llvm/bin/clang++
            cxx: /opt/rocm-5.2.0/llvm/bin/clang++
    roctracer-dev-api:
      buildable: false
      externals:
      - spec: roctracer-dev-api@5.2.0
        prefix: /opt/rocm-5.2.0/roctracer
        modules: [rocm/5.2.0]
    rocprim:
      buildable: false
      externals:
      - spec: rocprim@5.2.0
        prefix: /opt/rocm-5.2.0
        modules: [rocm/5.2.0]
    rocrand:
      buildable: false
      externals:
      - spec: rocrand@5.2.0
        prefix: /opt/rocm-5.2.0
        modules: [rocm/5.2.0]
    rocsolver:
      buildable: false
      externals:
      - spec: rocsolver@5.2.0
        prefix: /opt/rocm-5.2.0
        modules: [rocm/5.2.0]
    rocsparse:
      buildable: false
      externals:
      - spec: rocsparse@5.2.0
        prefix: /opt/rocm-5.2.0
        modules: [rocm/5.2.0]
    rocthrust:
      buildable: false
      externals:
      - spec: rocthrust@5.2.0
        prefix: /opt/rocm-5.2.0
        modules: [rocm/5.2.0]
    rocprofiler-dev:
      buildable: false
      externals:
      - spec: rocprofiler-dev@5.2.0
        prefix: /opt/rocm-5.2.0
        modules: [rocm/5.2.0]
  specs:
  - caliper@2.8.0 +rocm amdgpu_target=gfx90a
</pre></details>

<details><summary>Concretization</summary><pre>
==> Concretized caliper@2.8.0+rocm amdgpu_target=gfx90a
 -   el2m6vu  caliper@2.8.0%gcc@11.2.0+adiak~cuda~fortran+gotcha~ipo+libdw~libpfm+libunwind+mpi+papi+rocm+sampler+shared~sosflow amdgpu_target=gfx90a build_type=RelWithDebInfo arch=cray-sles15-zen3
[+]  cyoycfr      ^adiak@0.2.2%gcc@11.2.0~ipo+mpi+shared build_type=RelWithDebInfo arch=cray-sles15-zen3
[+]  ymt6u65          ^cmake@3.23.3%gcc@11.2.0~doc+ncurses+ownlibs~qt build_type=Release arch=cray-sles15-zen3
[+]  33sp27w              ^ncurses@6.3%gcc@11.2.0~symlinks+termlib abi=none arch=cray-sles15-zen3
[+]  pysfq3q                  ^pkgconf@1.8.0%gcc@11.2.0 arch=cray-sles15-zen3
[+]  riewo2i              ^openssl@1.1.1q%gcc@11.2.0~docs~shared certs=mozilla patches=3fdcf2d arch=cray-sles15-zen3
[+]  hql4jjz                  ^ca-certificates-mozilla@2022-07-19%gcc@11.2.0 arch=cray-sles15-zen3
[+]  ks6qvkk                  ^perl@5.34.1%gcc@11.2.0+cpanm+shared+threads arch=cray-sles15-zen3
[+]  kjdztqc                      ^berkeley-db@18.1.40%gcc@11.2.0+cxx~docs+stl patches=b231fcc arch=cray-sles15-zen3
[+]  rll2prr                      ^bzip2@1.0.8%gcc@11.2.0~debug~pic+shared arch=cray-sles15-zen3
[+]  3s2io65                          ^diffutils@3.8%gcc@11.2.0 arch=cray-sles15-zen3
[+]  po4cjxv                              ^libiconv@1.16%gcc@11.2.0 libs=shared,static arch=cray-sles15-zen3
[+]  yqxusqu                      ^gdbm@1.19%gcc@11.2.0 arch=cray-sles15-zen3
[+]  fdyfuat                          ^readline@8.1.2%gcc@11.2.0 arch=cray-sles15-zen3
[+]  ymk6b64                      ^zlib@1.2.12%gcc@11.2.0+optimize+pic+shared patches=0d38234 arch=cray-sles15-zen3
[+]  yfbzrpg          ^cray-mpich@8.1.16%gcc@11.2.0+wrappers arch=cray-sles15-zen3
[+]  f4zl2nt      ^elfutils@0.186%gcc@11.2.0+bzip2~debuginfod~nls+xz arch=cray-sles15-zen3
[+]  nlifhhv          ^m4@1.4.19%gcc@11.2.0+sigsegv patches=9dc5fbd,bfdffa7 arch=cray-sles15-zen3
[+]  42yhbvc              ^libsigsegv@2.13%gcc@11.2.0 arch=cray-sles15-zen3
[+]  7n4wqcj          ^xz@5.2.5%gcc@11.2.0+pic libs=shared,static arch=cray-sles15-zen3
[+]  7dkpqew      ^hip@5.2.0%gcc@11.2.0~ipo build_type=Release patches=959d1fe arch=cray-sles15-zen3
[+]  ykiof3l      ^hsa-rocr-dev@5.2.0%gcc@11.2.0+image~ipo+shared build_type=Release patches=71e6851 arch=cray-sles15-zen3
[+]  gldlvw2      ^libunwind@1.6.2%gcc@11.2.0~block_signals~conservative_checks~cxx_exceptions~debug~debug_frame+docs+pic+tests+weak_backtrace+xz~zlib components=none libs=shared,static arch=cray-sles15-zen3
[+]  hngdnef      ^llvm-amdgpu@5.2.0%gcc@11.2.0~ipo~link_llvm_dylib~llvm_dylib~openmp+rocm-device-libs build_type=Release patches=a08bbe1 arch=cray-sles15-zen3
[+]  qnzcb4x      ^papi@6.0.0.1%gcc@11.2.0~cuda+example~infiniband~lmsensors~nvml~powercap~rapl~rocm~rocm_smi~sde+shared~static_tools arch=cray-sles15-zen3
[+]  daysjb6      ^python@3.8.13%gcc@11.2.0+bz2+ctypes+dbm~debug+libxml2+lzma~nis~optimizations+pic+pyexpat+pythoncmd+readline+shared+sqlite3+ssl~tix~tkinter~ucs4+uuid+zlib patches=0d98e93,4c24573,ebdca64,f2fd060 arch=cray-sles15-zen3
[+]  kb64bk6          ^expat@2.4.8%gcc@11.2.0+libbsd arch=cray-sles15-zen3
[+]  6ii42bx              ^libbsd@0.11.5%gcc@11.2.0 arch=cray-sles15-zen3
[+]  nzueemp                  ^libmd@1.0.4%gcc@11.2.0 arch=cray-sles15-zen3
[+]  5nmtuqz          ^gettext@0.21%gcc@11.2.0+bzip2+curses+git~libunistring+libxml2+tar+xz arch=cray-sles15-zen3
[+]  yzo26m3              ^libxml2@2.9.13%gcc@11.2.0~python arch=cray-sles15-zen3
[+]  pa7goe7              ^tar@1.34%gcc@11.2.0 zip=pigz arch=cray-sles15-zen3
[+]  5yysicj                  ^pigz@2.7%gcc@11.2.0 arch=cray-sles15-zen3
[+]  4cs4kky                  ^zstd@1.5.2%gcc@11.2.0+programs compression=none libs=shared,static arch=cray-sles15-zen3
[+]  46izinn          ^libffi@3.4.2%gcc@11.2.0 arch=cray-sles15-zen3
[+]  4xmsjik          ^sqlite@3.39.2%gcc@11.2.0+column_metadata+dynamic_extensions+fts~functions+rtree arch=cray-sles15-zen3
[+]  pgmt2lr          ^util-linux-uuid@2.37.4%gcc@11.2.0 arch=cray-sles15-zen3
</pre></details>

Error:
```
$> spack -e . install
...
==> Installing caliper-2.8.0-el2m6vudox4wfakfarq4zmsszvjx3rl2
==> No binary for caliper-2.8.0-el2m6vudox4wfakfarq4zmsszvjx3rl2 found: installing from source
==> Fetching https://github.com/LLNL/Caliper/archive/v2.8.0.tar.gz
==> No patches needed for caliper
==> caliper: Executing phase: 'cmake'
==> caliper: Executing phase: 'build'
==> Error: ProcessError: Command exited with status 2:
    'make' '-j16'

16 errors found in build log:
     639    [ 98%] Linking CXX executable cali-query
     640    cd /tmp/lpeyrala/spack-stage/spack-stage-caliper-2.8.0-el2m6vudox4wfakfarq4zmsszvjx3rl2/spack-build-el2m6vu/src/tools/cali-query && /gpfs/alpine/csc439/world-shared/E4S/ParaTools/22.08/P
            rgEnv-gnu/spack/opt/spack/cray-sles15-zen3/gcc-11.2.0/cmake-3.23.3-ymt6u65ec5t5tt7lukz3gkzdyjr7h2yu/bin/cmake -E cmake_link_script CMakeFiles/cali-query.dir/link.txt --verbose=1
     641    /gpfs/alpine/csc439/world-shared/E4S/ParaTools/22.08/PrgEnv-gnu/spack/lib/spack/env/gcc/g++ -O2 -g -DNDEBUG "CMakeFiles/cali-query.dir/AttributeExtract.cpp.o" "CMakeFiles/cali-query.dir/
            cali-query.cpp.o" "CMakeFiles/query-common.dir/query_common.cpp.o" -o cali-query  -Wl,-rpath,/tmp/lpeyrala/spack-stage/spack-stage-caliper-2.8.0-el2m6vudox4wfakfarq4zmsszvjx3rl2/spack-bu
            ild-el2m6vu/src/tools/util:/tmp/lpeyrala/spack-stage/spack-stage-caliper-2.8.0-el2m6vudox4wfakfarq4zmsszvjx3rl2/spack-build-el2m6vu/src:/gpfs/alpine/csc439/world-shared/E4S/ParaTools/22.
            08/PrgEnv-gnu/spack/opt/spack/cray-sles15-zen3/gcc-11.2.0/adiak-0.2.2-cyoycfr5cteraicgoxjusf3gabdjemul/lib:/opt/cray/pe/mpich/8.1.16/ofi/gnu/9.1/lib::::::::::::::::::::::::::::::::::::::
            ::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
            ::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
            ::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
            ::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
            ::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
            ::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
            :::: ../util/libcaliper-tools-util.so.2.8.0 ../../libcaliper-serial.so.2.8.0 -pthread -Wl,-rpath-link,/gpfs/alpine/csc439/world-shared/E4S/ParaTools/22.08/PrgEnv-gnu/spack/opt/spack/cray
            -sles15-zen3/gcc-11.2.0/adiak-0.2.2-cyoycfr5cteraicgoxjusf3gabdjemul/lib:/opt/cray/pe/mpich/8.1.16/ofi/gnu/9.1/lib
     642    [100%] Linking CXX executable mpi-caliquery
     643    cd /tmp/lpeyrala/spack-stage/spack-stage-caliper-2.8.0-el2m6vudox4wfakfarq4zmsszvjx3rl2/spack-build-el2m6vu/src/tools/mpi-caliquery && /gpfs/alpine/csc439/world-shared/E4S/ParaTools/22.0
            8/PrgEnv-gnu/spack/opt/spack/cray-sles15-zen3/gcc-11.2.0/cmake-3.23.3-ymt6u65ec5t5tt7lukz3gkzdyjr7h2yu/bin/cmake -E cmake_link_script CMakeFiles/mpi-caliquery.dir/link.txt --verbose=1
     644    /gpfs/alpine/csc439/world-shared/E4S/ParaTools/22.08/PrgEnv-gnu/spack/lib/spack/env/gcc/g++ -O2 -g -DNDEBUG "CMakeFiles/mpi-caliquery.dir/mpi-caliquery.cpp.o" "../cali-query/CMakeFiles/q
            uery-common.dir/query_common.cpp.o" -o mpi-caliquery  -Wl,-rpath,/tmp/lpeyrala/spack-stage/spack-stage-caliper-2.8.0-el2m6vudox4wfakfarq4zmsszvjx3rl2/spack-build-el2m6vu/src:/tmp/lpeyral
            a/spack-stage/spack-stage-caliper-2.8.0-el2m6vudox4wfakfarq4zmsszvjx3rl2/spack-build-el2m6vu/src/tools/util:/opt/cray/pe/mpich/8.1.16/ofi/gnu/9.1/lib:/gpfs/alpine/csc439/world-shared/E4S
            /ParaTools/22.08/PrgEnv-gnu/spack/opt/spack/cray-sles15-zen3/gcc-11.2.0/adiak-0.2.2-cyoycfr5cteraicgoxjusf3gabdjemul/lib::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
            ::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
            ::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
            ::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
            ::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
            ::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
            :::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::: ../../libcaliper.so.2.8
            .0 ../util/libcaliper-tools-util.so.2.8.0 /opt/cray/pe/mpich/8.1.16/ofi/gnu/9.1/lib/libmpi_gnu_91.so /opt/cray/pe/mpich/8.1.16/ofi/gnu/9.1/lib/libmpi_gnu_91.so -Wl,-rpath-link,/gpfs/alpi
            ne/csc439/world-shared/E4S/ParaTools/22.08/PrgEnv-gnu/spack/opt/spack/cray-sles15-zen3/gcc-11.2.0/adiak-0.2.2-cyoycfr5cteraicgoxjusf3gabdjemul/lib
  >> 645    /usr/bin/ld: ../../libcaliper-serial.so.2.8.0: undefined reference to `hipKernelNameRef'
  >> 646    /usr/bin/ld: ../../libcaliper-serial.so.2.8.0: undefined reference to `hipKernelNameRefByPtr'
  >> 647    collect2: error: ld returned 1 exit status
  >> 648    make[2]: *** [src/tools/cali-stat/CMakeFiles/cali-stat.dir/build.make:102: src/tools/cali-stat/cali-stat] Error 1
     649    make[2]: Leaving directory '/tmp/lpeyrala/spack-stage/spack-stage-caliper-2.8.0-el2m6vudox4wfakfarq4zmsszvjx3rl2/spack-build-el2m6vu'
  >> 650    make[1]: *** [CMakeFiles/Makefile2:1652: src/tools/cali-stat/CMakeFiles/cali-stat.dir/all] Error 2
     651    make[1]: *** Waiting for unfinished jobs....
  >> 652    /usr/bin/ld: ../../libcaliper-serial.so.2.8.0: undefined reference to `hipKernelNameRef'
  >> 653    /usr/bin/ld: ../../libcaliper-serial.so.2.8.0: undefined reference to `hipKernelNameRefByPtr'
  >> 654    collect2: error: ld returned 1 exit status
  >> 655    make[2]: *** [src/tools/cali-query/CMakeFiles/cali-query.dir/build.make:120: src/tools/cali-query/cali-query] Error 1
     656    make[2]: Leaving directory '/tmp/lpeyrala/spack-stage/spack-stage-caliper-2.8.0-el2m6vudox4wfakfarq4zmsszvjx3rl2/spack-build-el2m6vu'
  >> 657    make[1]: *** [CMakeFiles/Makefile2:1625: src/tools/cali-query/CMakeFiles/cali-query.dir/all] Error 2
  >> 658    /usr/bin/ld: ../../libcaliper.so.2.8.0: undefined reference to `hipKernelNameRef'
  >> 659    /usr/bin/ld: ../../libcaliper.so.2.8.0: undefined reference to `hipKernelNameRefByPtr'
  >> 660    collect2: error: ld returned 1 exit status
  >> 661    make[2]: *** [src/tools/mpi-caliquery/CMakeFiles/mpi-caliquery.dir/build.make:106: src/tools/mpi-caliquery/mpi-caliquery] Error 1
     662    make[2]: Leaving directory '/tmp/lpeyrala/spack-stage/spack-stage-caliper-2.8.0-el2m6vudox4wfakfarq4zmsszvjx3rl2/spack-build-el2m6vu'
  >> 663    make[1]: *** [CMakeFiles/Makefile2:1680: src/tools/mpi-caliquery/CMakeFiles/mpi-caliquery.dir/all] Error 2
     664    make[1]: Leaving directory '/tmp/lpeyrala/spack-stage/spack-stage-caliper-2.8.0-el2m6vudox4wfakfarq4zmsszvjx3rl2/spack-build-el2m6vu'
  >> 665    make: *** [Makefile:139: all] Error 2
```


### Error message

<details><summary>Error message</summary><pre>
... see above
</pre></details>


### Information on your system

* **Spack:** 0.19.0.dev0 (fce58626df2a4f526de0771adfa8b7afb673a361)
* **Python:** 3.9.7
* **Platform:** cray-sles15-zen3
* **Concretizer:** clingo

### Additional information

[CMakeOutput.log](https://github.com/spack/spack/files/9648976/CMakeOutput.log)
[spack-build-env.txt](https://github.com/spack/spack/files/9648977/spack-build-env.txt)
[spack-build-out.txt](https://github.com/spack/spack/files/9648978/spack-build-out.txt)

@daboehme @wspear @lukebroskop 

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

--- a/var/spack/repos/builtin/packages/caliper/package.py
+++ b/var/spack/repos/builtin/packages/caliper/package.py
@@ -138,6 +138,7 @@ class Caliper(CMakePackage, CudaPackage, ROCmPackage):
             args.append("-DCUPTI_PREFIX=%s" % spec["cuda"].prefix)
 
         if "+rocm" in spec:
+            args.append("-DCMAKE_CXX_COMPILER={0}".format(spec["hip"].hipcc))
             args.append("-DROCM_PREFIX=%s" % spec["hsa-rocr-dev"].prefix)
 
         return args

